scalanet/discovery/test/src/io/iohk/scalanet/discovery/crypto/SigAlgSpec.scala

```diff
@@ -0,0 +1,8 @@
+package io.iohk.scalanet.discovery.crypto
+
+import org.scalatest._
+
+class SigAlgSpec extends FlatSpec with Matchers {
+  // Use test vectors from https://wizardforcel.gitbooks.io/practical-cryptography-for-developers-book/content/digital-signatures/ecdsa-sign-verify-examples.html
+  // Implement recovery based on https://github.com/ConsenSysMesh/cava/blob/master/crypto/src/main/java/net/consensys/cava/crypto/SECP256K1.java
```

Comment:
hm we already has recovery in mantis and whole (ECDSA key handling) is this somewhat better ?

Reply:
Yes I realised later that it must be available in Mantis, just left the comment as a reminder in case it's not. What I was planning to do is provide the implementation at the time when we instantiate this dependency within Mantis. Talked to @mirkoAlic yesterday about extracting RLP and Crypto; he's of the opinion that it should be polished a bit before we do it and it would be 1-2 weeks for just RLP.

Reply:
At the time I was looking at whether [CryptoUtils](https://github.com/input-output-hk/scalanet/blob/develop/scalanet/src/io/iohk/scalanet/crypto/CryptoUtils.scala) could be used to implement the `SigAlg`.